src/commands/githubActions/bumpUpstream/setup.ts

```diff
@@ -0,0 +1,99 @@
+    const envFileArgs = readBuildSdkEnvFileNotThrow(dir);
+
+    const { manifest, format } = readManifest({ dir });
+    const compose = readCompose({ dir });
+
+    const upstreamRepos = envFileArgs
+        ? [envFileArgs._BUILD_UPSTREAM_REPO]
+        : parseCsv(manifest.upstreamRepo);
+    const upstreamArgs = envFileArgs
+        ? [envFileArgs._BUILD_UPSTREAM_VERSION]
+        : parseCsv(manifest.upstreamArg || "UPSTREAM_VERSION");
+
+    // Create upstream settings after validation
+    validateUpstreamData(upstreamRepos, upstreamArgs);
+    const upstreamSettings = upstreamRepos.map((repo, i) => ({
+        upstreamRepo: repo,
+        upstreamArg: upstreamArgs[i],
+    }));
+
+    const gitSettings = getGitSettings();
+
+    const ethProviders = getEthProviders(useFallback, userEthProvider);
+
+    const ethProviderAvailable = await getFirstAvailableEthProvider({
+        providers: ethProviders
+    });
+
+    if (!ethProviderAvailable)
+        throw Error(`No eth provider available. Tried: ${ethProviders.join(", ")}`);
+
+    return {
+        upstreamSettings,
+        manifestData: { manifest, format },
+        compose,
+        gitSettings,
+        ethProvider: ethProviderAvailable,
+    };
+}
+
+function getEthProviders(useFallback: boolean, userEthProvider: string): string[] {
+    const defaultEthProviders = ["remote", "infura"];
+
+    const ethProviders = useFallback
+        ? [userEthProvider, ...defaultEthProviders.filter(p => p !== userEthProvider)]
+        : [userEthProvider];
+
+    return ethProviders;
+}
+
+function getGitSettings(): GitSettings {
+    const githubActor = process.env.GITHUB_ACTOR || "bot";
+    const userEmail = `${githubActor}@users.noreply.github.com`;
+
+    return {
+        userName: githubActor,
+        userEmail
+    };
+}
+
+export function validateUpstreamData(upstreamRepos: string[], upstreamArgs: string[]): void {
+    if (upstreamRepos.length < 1)
+        throw new Error("Must provide at least one 'upstream_repo'");
+
+    if (upstreamRepos.length !== upstreamArgs.length)
+        throw new Error(`'upstream_repo' must have the same length as 'upstream_argNames'. Got ${upstreamRepos.length} repos and ${upstreamArgs.length} args.`);
+
+    if (!arrIsUnique(upstreamRepos))
+        throw new Error("upstreamRepos not unique");
+
+    if (!arrIsUnique(upstreamArgs))
+        throw new Error("upstreamArgs not unique");
+}
+
+
+export function printSettings(upstreamSettings: UpstreamSettings[], gitSettings: GitSettings, manifest: Manifest, compose: Compose, ethProvider: string): void {
+
+    console.log(`
+  
+    Upstream Settings - ${JSON.stringify(upstreamSettings, null, 2)}
+  
+    Git Settings - ${JSON.stringify(gitSettings, null, 2)}
+    
+    Manifest - ${JSON.stringify(manifest, null, 2)}
+    
+    Compose - ${JSON.stringify(compose, null, 2)}
+    
+    ETH Provider - ${ethProvider}
```

Comment:
would be interesting to print also the ipfs provider for debugging

Reply:
It is not possible to pass a flag for the IPFS provider in this command. It is hardcoded in the `buildAndComment` function:

```
  const { releaseMultiHash } = await buildHandler({
    provider: "pinata",
    upload_to: "ipfs",
    require_git_data: true,
    delete_old_pins: true,
    verbose: true
  });
  ```

So it is not possible to print the IPFS provider here